Objects/genericaliasobject.c

```diff
@@ -180,14 +180,16 @@ tuple_extend(PyObject **dst, Py_ssize_t dstindex,
 PyObject *
 _Py_make_parameters(PyObject *args)
 {
-    Py_ssize_t nargs = PyTuple_GET_SIZE(args);
+    assert (PyTuple_Check(args) || PyList_Check(args));
+    Py_ssize_t nargs = PySequence_Length(args);
```

Comment:
This is unsafe because it invokes `__len__`, which may be a lie on subclasses of list/tuple.

```
>>> class Liar(tuple):
...     def __len__(self): return 42
...     
>>> type X[**P] = P
>>> from typing import TypeVar
>>> T = TypeVar("T")
>>> X2=X[[Liar((T,))]]
... 
>>> X2[int]
zsh: segmentation fault  ./python.exe
```